### PR TITLE
Carved out HACKING.md from README.md

### PR DESCRIPTION
**Pull request by UtkarshBhatthere**
Created a separate document for steps related to building ceph-rocks and doing script based deployment in local testing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,135 +1,33 @@
-# Ubuntu Ceph Container Images
-Home to Ubuntu based Ceph container Images.
+# Ceph ROCKs
+Home to Ubuntu based Ceph OCI Images.
```
Comment (sabaini) [inline README.md:2]: Minor suggestion: could add a link to explain OCI, e.g. to https://en.wikipedia.org/wiki/Open_Container_Initiative , either here or below

File: README.md [hunk 2]
```diff
@@ -1,135 +1,33 @@
...
-All images available locally can be checked through:
-```
-$ sudo docker images
-REPOSITORY         TAG          IMAGE ID       CREATED       SIZE
-canonical/ceph     latest       4d47f598e7ef   4 hours ago   1.51GB
-```
-
-This freshly baked Image can now be used for deploying Ceph using:
+## Using Ceph ROCKs
+The Ceph ROCK available at [GH Container Repository](https://github.com/canonical/ceph-containers/pkgs/container/ceph) can be used with popular containerised-ceph deployment tools like:
```
Comment (sabaini) [inline README.md:11]: s/at/at the/

File: README.md [hunk 3]
```diff
@@ -1,135 +1,33 @@
...
-<qualified_image_name> can be a reference to a container image hosted on any public container registry.
-For Example:
-```
-$ python3 test/deploy.py image ghcr.io/canonical/ceph:main
-```
-2.) Use Script to clean a deployment (using script generated model file):
-```
-python3 test/deploy.py delete model-88HJ.json
-```
+A green light means that a particular feature is supported, and a red light means that it isn't. A yellow light means that a feature is _technically_ supported, but needs a considerable amount of fiddling.
```
Comment (sabaini) [inline README.md:20]: I realize this has just been moved but still wonder about phrasing here, maybe "needs additional manual configuration"?

File: HACKING.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+## Table of Contents
+1. [Introduction to ROCKs](#introduction-to-rocks)
+2. [Tools](#tools)
+3. [References](#references)
+4. [Build Guide](#build-guide)
+6. [Uisng Ceph ROCKs](#using-ceph-rocks)
+
+## Introduction to ROCKs
+> A [ROCK](https://canonical-rockcraft.readthedocs-hosted.com/en/latest/explanation/rocks/) is an OCI compliant container image archive built on top of Ubuntu LTS releases. ROCKs are interoperable with OCI-compliant tools and platforms.
```
Comment (sabaini) [inline HACKING.md:12]: Maybe here as well a link to explain "OCI", e.g. https://en.wikipedia.org/wiki/Open_Container_Initiative

File: HACKING.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+## Tools
+ROCKs are built using a snap called [rockcraft](https://github.com/canonical/rockcraft). It uses [lxd](https://snapcraft.io/lxd) to pull dependencies and build an artefact completely isolated from the host system. This makes it easier for developers to work on ROCKs without polluting their host system with unwanted dependencies.
+You can install snapcraft and lxd using snap tool.
+```bash
+sudo snap install rockcraft --classic
+sudo snap install lxd
+```
+
+> **_NOTE:_**
+For a detailed how-to-use snapcraft tool guide, check-out [ROCK Docs](https://canonical-rockcraft.readthedocs-hosted.com/en/latest/)
```
Comment (sabaini) [inline HACKING.md:25]: s/snapcraft/rockcraft/ I presume?

File: HACKING.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+    Kubernetes provides a command line tool for communicating with the cluster's control plane, using the Kubernetes API. This tool is named `kubectl`. Kubectl is built using [kubectl.go](kubectl/kubectl.go) and `go` snap.
+
+4. **[local-files](rockcraft.yaml#L93)**
+
+    Contains bash scripts, configuration files (s3cfg, ceph.defaults, confd templates, etc) that go in the artefact.
+
+5. **[ceph-container-service](rockcraft.yaml#L19)**
+
+    ROCKs use [pebble](https://github.com/canonical/pebble) as the official entrypoint of the OCI artefact. However, orchestration tools like [Rook](https://rook.io/) and [Cephadm](https://docs.ceph.com/en/latest/cephadm/) don't rely on the configured entrypoint and overwrite it when spawning containers. Hence, the pebble entrypoint is an experimental feature that has not been tested for production deployments.
```
Comment (sabaini) [inline HACKING.md:49]: I wonder about this para, I feel it lacks a bit of context what `ceph-container-service` actually is?

**Review by sabaini — CHANGES_REQUESTED** [review]
Hi Utkarsh, looks good in general. I have added some minor suggestions/commentary inline. Thanks!

File: README.md (hunk 1 above)
Comment (UtkarshBhatthere) [reply README.md:2]: +1

File: README.md (hunk 3 above)
Comment (UtkarshBhatthere) [reply README.md:20]: I have changed to "needs operator intervention"

File: HACKING.md (hunk 5 above)
Comment (UtkarshBhatthere) [reply HACKING.md:25]: YES! So many crafts 😄

File: HACKING.md (hunk 6 above)
Comment (UtkarshBhatthere) [reply HACKING.md:49]: Added description of the service.

**Review by sabaini — APPROVED** [review]

**Pull request merged**